### PR TITLE
Configure community repo submit-queue.

### PR DESCRIPTION
**Pull request by cjwagner**
Depends on kubernetes/community#842

### CODE REVIEW

**Comment by cjwagner** [comment]
/cc @rmmh 
@grodrigues3 

I will deploy with:
```bash
TARGET=community READONLY=false make deploy && \
k delete cm nginx-https-redirect && \
k create cm nginx-https-redirect --from-file=nginx.conf=submit-queue/nginx-redirect/nginx.conf && \
k delete po nginx-https-redirect-113117905-b8wv0 && \
k apply --record -f submit-queue/ingress.yaml
```

**Comment by fejta** [comment]
Can we put a Makefile that runs those commands in the `mungegithub/submit-queue/deployment/community/` dir?

**Comment by grodrigues3** [comment]
@cjwagner Thanks!  I'll let Ryan review/lgtm.

File: mungegithub/submit-queue/nginx-redirect/nginx.conf [hunk 1]
```diff
@@ -5,6 +5,23 @@ events {
...
+    server_name community.submit-queue.k8s.io;
+    location / {
+        if ($http_user_agent ~ (GoogleHC) ) {
+          return 200;
+        }
+        if ($http_x_forwarded_proto != 'https') {
+          return 301 https://$host$request_uri;
+        }
+        proxy_pass http://community-sq-status;
+        proxy_set_header Host $host;
```
Comment (rmmh) [inline mungegithub/submit-queue/nginx-redirect/nginx.conf]: can you add another commit to hoist these proxy_set_header directives up to the http scope? should reduce duplication in this file

**Comment by fejta** [comment]
Also:

a) Create a Makefile in mungegithub/submit-queue/deployment/ that defines a command to update the submit queues in all the repos
b) Add a unit test to validate that mungegithub/submit-queue/deployment/REPO has a Makefile and mungegithub/submit-queue/deployment/community/ will find and update every REPO?

**Comment by cjwagner** [comment]
@fejta I'll make those additions in a separate PR.

**Comment by grodrigues3** [comment]
Err, if the SQ is running on community will we require OWNERS files too?

**Comment by grodrigues3** [comment]
Addresses https://github.com/kubernetes/test-infra/issues/3675

**Comment by cjwagner** [comment]
> Err, if the SQ is running on community will we require OWNERS files too?

OWNERS files are needed to use the approval-handler munger which isn't enabled in the community repo config currently. However, we ultimately want the same mungers to run on every repo so now would be a good time to enable that and add OWNERS files.

**Comment by grodrigues3** [comment]
> OWNERS files are needed to use the approval-handler munger which isn't enabled in the community repo config currently

Blunderbuss also uses the OWNERS files for assigning reviewers.  And the submit queue requires the approved and lgtm labels in order to merge things right?  

- get OWNERS files
- add the approved and lgtm labels (part of #2504 )

Anything else I'm missing before we can merge/deploy?

**Comment by rmmh** [comment]
It might make sense for labels like lgtm/approved to be an explicit part of the SQ config, instead of buried in the code.

**Comment by grodrigues3** [comment]
I think there is a flag to disable merge on approved label in the SQ.

@rmmh do you mean that the dependency on lgtm/approved should be captured in the SQ config, such that the labels are auto created if they don't exist already?  

Or were you saying there should be a flag like `gate-merge-on` that takes the list of labels required for merge?

**Comment by cjwagner** [comment]
> Blunderbuss also uses the OWNERS files for assigning reviewers. And the submit queue requires the approved and lgtm labels in order to merge things right?
> 
>   - get OWNERS files
>   - add the approved and lgtm labels (part of #2504 )
> 
> Anything else I'm missing before we can merge/deploy?

The correct labels already exist. Blunderbuss uses OWNERS files, but it isn't enabled in the community config right now. The submit-queue also isn't configured to require the approved label right now. I'll update this PR to enable Blunderbuss and gate on the approved label.
I think thats all there is to do before merging and deploying unless you are want to enable more mungers (lgtm-after-commit, comment-deleter, close-stale, release-note-label, etc).

**Comment by cjwagner** [comment]
@grodrigues3 I enabled the following mungers: [blunderbuss, approval-handler, lgtm-after-commit, comment-deleter] to try to maintain consistency across repos. Is this ok with you?

**Comment by grodrigues3** [comment]
Fixes https://github.com/kubernetes/community/issues/201

**Comment by grodrigues3** [comment]
let me get the owners files PR merged before merging turning this on

File: mungegithub/submit-queue/deployment/community/configmap.yaml [hunk 2]
```diff
@@ -19,9 +23,13 @@ protected-branches: "master"
 nonblocking-jenkins-jobs: ""
 do-not-merge-milestones: ""
 admin-port: 9999
-context-url: ""
+context-url: https://community.submit-queue.k8s.io
+gate-cla: true
+gate-approved: true
```
Comment (grodrigues3) [inline mungegithub/submit-queue/deployment/community/configmap.yaml]: let's set this to false

File: mungegithub/submit-queue/deployment/community/configmap.yaml [hunk 3]
```diff
@@ -2,7 +2,11 @@
 http-cache-dir: /cache/httpcache
 organization: kubernetes
 project: community
-pr-mungers: size, submit-queue,needs-rebase, sig-mention-handler, 
+# Make sure approval-handler and blunderbuss run before submit-queue.
+# Otherwise it's going to take an extra-cycle to detect the label change.
+# Run blunderbuss before approval-handler, so that we can suggest approvers
+# based on assigned reviewer.
+pr-mungers: blunderbuss, approval-handler, size, submit-queue, needs-rebase, sig-mention-handler, lgtm-after-commit, comment-deleter
```
Comment (grodrigues3) [inline mungegithub/submit-queue/deployment/community/configmap.yaml]: turn off approval-handler for now

Comment (cjwagner) [inline mungegithub/submit-queue/deployment/community/configmap.yaml]: done

File: mungegithub/submit-queue/deployment/community/configmap.yaml (hunk 2 above)
Comment (cjwagner) [inline mungegithub/submit-queue/deployment/community/configmap.yaml]: done

**Pull request merged**